File: lib/es6/js_promise2.js



let then = (function(p, cont) {
    return Promise.resolve(p).then(cont)
  });

let $$catch = (function(p, cont) {
      return Promise.resolve(p).catch(cont)
    });

export {
  then,
  $$catch,
}
/* No side effect */

File: jscomp/others/js_promise2.ml
include Js_promise

let then_ : 'a promise -> ('a -> 'b promise) -> 'b promise =
  [%raw {|
  function(p, cont) {
    Promise.resolve(p).then(cont)
  }
  |}]

let catch_ : 'a promise -> (error -> 'a promise) -> 'a promise =
  [%raw
    {|
    function(p, cont) {
      Promise.resolve(p).catch(cont)
    }
    |}]
